### PR TITLE
Add missing legality rules for async block expressions

### PR DESCRIPTION
**Pull request by Veykril**
Closes https://github.com/ferrocene/specification/issues/98

### CODE REVIEW

File: src/expressions.rst [hunk 1]
```diff
@@ -723,20 +723,23 @@ async Blocks
...
-An :t:`async block expression` denotes a new :t:`control flow boundary`.
+An :t:`async block expression` denotes a new :t:`async control flow boundary`.
+
+:dp:`fls_eodEo8qRE0JH`
+An :t:`async block expression` is subject to :t:`capturing`.
 
 :dp:`fls_ncd0wkgtldem`
-The :t:`type` of an :t:`async block expression` shall implement the
-:std:`core::future::Future` trait.
+The :t:`type` of an :t:`async block expression` is an anonymous unique :t:`type`
```
Comment (kirtchev-adacore) [inline src/expressions.rst:738]: "unique anonymous" sounds a little better.

File: src/expressions.rst [hunk 2]
```diff
@@ -3419,7 +3422,7 @@ Closure Expressions
 
 :dp:`fls_2d141c9a0yui`
 A :t:`closure expression` is an :t:`expression` that defines a :t:`closure
-type`.
+type` and constructs a value of such a :t:`type`.
```
Comment (kirtchev-adacore) [inline src/expressions.rst:3425]: The following might sounds a bit better:

A closure expression is an expression that defines a closure type and constructs a value of that type.

File: src/glossary.rst [hunk 3]
```diff
@@ -412,11 +412,20 @@ async block expression
...
 with :t:`keyword` ``async`` and encapsulates behavior which is executed in
-asynchronous manner.
+an asynchronous manner.
 
 :dp:`fls_je689rormhd6`
 See :s:`AsyncBlockExpression`.
 
+.. _fls_lYrTaCM1LcXU:
+
+control flow boundary
```
Comment (kirtchev-adacore) [inline src/glossary.rst:422]: `async` is missing.

File: src/glossary.rst [hunk 4]
```diff
@@ -412,11 +412,20 @@ async block expression
...
+^^^^^^^^^^^^^^^^^^^^^
+
+:dp:`fls_EXoGOkCRsfKK`
+An :dt:`async control flow boundary` is a :t:`control flow boundary` that
+additionally allows suspending the execution via :t:`[await expression]s`.
```
Comment (kirtchev-adacore) [inline src/glossary.rst:427]: How about

An async control flow boundary is a control flow boundary that allows the suspension of execution via await expressions.

File: src/glossary.rst [hunk 5]
```diff
@@ -1031,7 +1040,7 @@ closure expression
 
 :dp:`fls_x87rhn9ikz00`
 A :dt:`closure expression` is an :t:`expression` that defines a :t:`closure
-type`.
+type` and constructs a value of such a :t:`type`.
```
Comment (kirtchev-adacore) [inline src/glossary.rst:1043]: Might need to be synced with the text in Expressions.

File: src/expressions.rst [hunk 6]
```diff
@@ -4490,20 +4493,8 @@ A :t:`future operand` is an :t:`operand` whose :t:`future` is being awaited by
...
-asynchronous control flow.
-
-:dp:`fls_Rr2Wl6mp4oXS`
-The following constructs are :t:`[async control flow boundary]ies`:
-
-* :dp:`fls_xilkqy5piyh0`
-  An :t:`async block expression`.
-
-* :dp:`fls_cr61i8so7cty`
-  The :t:`function body`` of an :t:`async function`.
```
Comment (kirtchev-adacore) [inline src/expressions.rst:4506]: We are tossing these out? I don't think we have specified which constructs constitute a `control flow boundary`...

**Review by kirtchev-adacore — CHANGES_REQUESTED** [review]

File: src/expressions.rst (hunk 6 above)
Comment (Veykril) [reply src/expressions.rst:4506]: They are specified in the corresponding sections, see https://github.com/ferrocene/specification/pull/253/files#diff-302b8fbe696da8535fb26eb2db6211010f45141514fefe8e0e071be8c2ed5b50R100-R102 and https://github.com/ferrocene/specification/pull/253/files#diff-d2497d8980a80566992fa05d12234586c1006c2261a3e39e33f1b50703fc8701R731-R732

Comment (kirtchev-adacore) [reply src/expressions.rst:4506]: OK!

**Review by kirtchev-adacore — APPROVED** [review]

**Comment by kirtchev-adacore** [comment]
bors merge

**Pull request merged**